### ISSUE TITLE
PruneObjectOutputHandler can be unusably slow for complex parent ROIs

### ISSUE THREAD

**Issue by petebankhead** [qupath/qupath-extension-instanseg#88]
I found InstanSeg to be unusable with a large and complex parent ROI generated using a threshold.

The troublesome line seems to be
https://github.com/qupath/qupath-extension-instanseg/blob/2f5ea38f85deeee9ff9473dd49a4637b9b8f1f89/src/main/java/qupath/ext/instanseg/core/PruneObjectOutputHandler.java#L66

This isn't necessarily the extensions fault, and maybe `PixelProcessorUtils.maskObject` needs to be changed - but I'm logging the issue here, since it's the place people are most likely to encounter it.

Basically, object masking by intersection can be _much_ too expensive with complex parent objects, and we need to find ways to simplify this.

**Comment by alanocallaghan** [comment]
What does "unusably close" mean?

**Comment by petebankhead** [comment]
That I'm too tired... edited now.

**Comment by petebankhead** [comment]
I think with this PR in the main repo, geometry operations are no longer the bottleneck: https://github.com/qupath/qupath/pull/1639
